libfwupd/fwupd-device-private.h

```diff
@@ -15,6 +15,7 @@ G_BEGIN_DECLS
 
 FwupdDevice	*fwupd_device_from_variant		(GVariant	*data);
 GVariant	*fwupd_device_to_variant		(FwupdDevice	*device);
+GVariant	*fwupd_trusted_device_to_variant	(FwupdDevice	*device);
```

Comment:
The way we've done this in the past is to add a `fwupd_device_to_variant_full()` function, that has both the `FwupdDevice	*device` and a bitfield we can extend in the future without breaking API. We could then pass something like `FWUPD_DEVICE_FLAG_TRUSTED` to get all the properties.